### PR TITLE
Source S3: fix discovery issues

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/1664
https://github.com/airbytehq/oncall/issues/1652

## How
This PR actually fixes two different problems:
- Return empty schema if no files found during discovery
- Do not infer data types when user schema is applied (see more in code comment)

## 🚨 User Impact 🚨
The second fix will only impact connections with:
 - File type = CSV and 
 - User enforced schema and
 - Large files and small block size
 
 Some data may change it's type but still it shouldn't break syncs (see more in code comment)

### CODE REVIEW

**Review by bnchrch — APPROVED** [review]
One small nit but otherwise LGTM.



Friendly reminder to Run CAT before publishing.

File: airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def _get_schema_dict_without_inference(self, file: Union[TextIO, BinaryIO]) -> M
         quote_char = self.format.quote_char
         reader = csv.reader([six.ensure_text(file.readline())], delimiter=delimiter, quotechar=quote_char)
         field_names = next(reader)
+        file.seek(0)
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py]: Added a comment

**Pull request merged**